R/mainTests.r

```diff
@@ -1153,6 +1160,454 @@ country_code <- function(data) {
+                             <br> - Every school name should have only one URN and vice versa."),
+          "result" = "FAIL"
+        )
+      } else {
+        if (length(multi_count_code) == 1 & length(multi_count_name) == 1) {
+          output <- list(
+            "message" = paste0("The following school_name has multiple URNs '", paste0(multi_count_code), "', 
+                              <br> and the following URN has multiple names '", paste0(multi_count_name), "'.
+                             <br> - Every school name should have only one URN and vice versa."),
+            "result" = "FAIL"
+          )
+        } else {
+          if (length(multi_count_code) > 1 & length(multi_count_name) == 1) {
+            output <- list(
+              "message" = paste0("The following school_names have multiple URNs '", paste0(multi_count_code, collapse = "', '"), "', 
+                              <br> and the following URN has multiple names '", paste0(multi_count_name), "'.
+                             <br> - Every school name should have only one URN and vice versa."),
+              "result" = "FAIL"
+            )
+          } else {
+            if (length(multi_count_code) == 1 & length(multi_count_name) > 1) {
+              output <- list(
+                "message" = paste0("The following school_name has multiple URNs '", paste0(multi_count_code), "', 
+                              <br> and the following URNs have multiple names '", paste0(multi_count_name, collapse = "', '"), "'.
+                             <br> - Every school name should have only one URN and vice versa."),
+                "result" = "FAIL"
+              )
+            } else {
+              if (length(multi_count_code) == 0 & length(multi_count_name) > 1) {
+                output <- list(
+                  "message" = paste0("The following URNs have multiple names '", paste0(multi_count_name, collapse = "', '"), "'.
+                             <br> - Every school name should have only one URN and vice versa."),
+                  "result" = "FAIL"
+                )
+              } else {
+                output <- list(
+                  "message" = paste0("The following school_names have multiple codes '", paste0(multi_count_code, collapse = "', '"), "', 
+                              <br> and the following URNs have multiple names '", paste0(multi_count_name, collapse = "', '"), "'.
+                             <br> - Every school name should have only one URN and vice versa."),
+                  "result" = "FAIL"
+                )
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  return(output)
+}
+
+
+# other_geography_duplicates  ----------------------------------------
+# check that there is a 1:1 relationship between geography codes and names
+# ignore school
+lower_level_geog_names <- geography_matrix[c(7:12, 14:16), 2:3] %>% as.character()
```

Comment:
Pretty nifty, didn't realise you could use `as.character()` to convert into a vector